src/parsing/military_parsing.cpp

```diff
@@ -110,75 +110,79 @@ void make_unit(std::string_view name, token_generator& gen, error_handler& err,
 		context.state.military_definitions.irregular = new_id;
 		err.accumulated_warnings += "Fallbacking to detecting an irregular from name " + err.file_name + "\n";
 	}
+	if(!bool(context.state.military_definitions.artillery) && name == "artillery") {
```

Comment:
But ... you didn't remove the way artillery is actually detected, meaning that this assignment won't do anything